ui/src/routes/_layout/organizations/$orgId/products/$productId/repositories/$repoId/_layout/runs/$runIndex/rule-violations/index.tsx

```diff
@@ -18,26 +18,236 @@
+      <DialogHeader>
+        <DialogTitle>{details.ruleViolation.rule}</DialogTitle>
+        <DialogDescription>{details?.ruleViolation.message}</DialogDescription>
+      </DialogHeader>
+      <div className='grid grid-cols-8 gap-2 text-sm'>
+        <div className='col-span-2 font-semibold'>License:</div>
+        <div className='col-span-6'>{details.ruleViolation.license}</div>
+        <div className='col-span-2 font-semibold'>License source:</div>
+        <div className='col-span-6'>{details.ruleViolation.licenseSource}</div>
+        <div className='col-span-2 font-semibold'>How to fix:</div>
+      </div>
+      <div className='flex whitespace-pre-line break-all text-sm'>
+        {details.ruleViolation.howToFix}
+      </div>
+      <DialogFooter>
+        <DialogClose asChild>
+          <Button type='button' variant='secondary'>
+            Close
+          </Button>
+        </DialogClose>
+      </DialogFooter>
+    </DialogContent>
+  );
+};
 
 const RuleViolationsComponent = () => {
+  const params = Route.useParams();
+  const search = Route.useSearch();
+
+  // Memoize the search parameters to prevent unnecessary re-rendering
+
+  const pageIndex = useMemo(
+    () => (search.page ? search.page - 1 : 0),
+    [search.page]
+  );
+
+  const pageSize = useMemo(
+    () => (search.pageSize ? search.pageSize : defaultPageSize),
+    [search.pageSize]
+  );
+
+  const groups = useMemo(
+    () => (search.groups ? search.groups : ['Severity']),
+    [search.groups]
+  );
+
+  const { data: ortRun } = useRepositoriesServiceGetOrtRunByIndexSuspense({
+    repositoryId: Number.parseInt(params.repoId),
+    ortRunIndex: Number.parseInt(params.runIndex),
+  });
+
+  const { data: ruleViolations } =
+    useRuleViolationsServiceGetRuleViolationsByRunIdSuspense({
+      runId: ortRun.id,
+      // Fetch all data at once, as we need to do both grouping and
+      // pagination in front-end for consistency in data handling.
+      limit: 100000,
+      sort: 'rule',
+    });
+
+  const table = useReactTable({
+    data: ruleViolations?.data || [],
+    columns,
+    state: {
+      pagination: {
+        pageIndex,
+        pageSize,
+      },
+      grouping: groups,
+    },
+    getCoreRowModel: getCoreRowModel(),
+    getExpandedRowModel: getExpandedRowModel(),
+    getGroupedRowModel: getGroupedRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+  });
+
   return (
     <Card className='h-fit'>
       <CardHeader>
-        <CardTitle>Rule violations</CardTitle>
-        <CardDescription>Rule violations of the project.</CardDescription>
+        <CardTitle>Rule violations (ORT run global ID: {ortRun.id})</CardTitle>
+        <CardDescription>
+          These are the rule violations found currently in the project. Please
+          note that the status may change over time, as your project
+          dependencies change. Therefore, your project should be scanned for
+          rule violations regularly.
```

Comment:
The text may be a little misleading, as these are not rule violations found currently in the project, but found in the specific ORT run. I'd modify the text accordingly.